src/jsifier.js

```diff
@@ -45,7 +47,8 @@ function JSify(data, functionsOnly) {
     // else. This lets us not hold any strings in memory, we simply print
     // things out as they are ready.
 
-    var shellParts = read(shellFile).split('{{BODY}}');
+    var thirdParty = thirdPartyFiles.map(function(f) { return read(f) }).join('\n');
+    var shellParts = read(shellFile).replace('{{THIRD_PARTY}}', thirdParty).split('{{BODY}}');
```

Comment:
@juj and @kripken, I just committed my functions from above in place of sodiumutil, which I think we'll be okay with (particularly if we get that MEM_INIT_METHOD test working here).

As far as the large byte array problem with decoding UTF-8: upon further investigation, apparently the issue was that libsodium's original `to_string` used `decodeURIComponent(escape(String.fromCharCode.apply(null, bytes)))` instead of a `String.fromCharCode(bytes[i])` for loop as I had here. Using the former method, an exception would be thrown with byte arrays larger than a few hundred KB, but the latter is limited only by the heap size as far as I can tell.

The reason it was implemented that way was for better error handling; the latter method will happily return weird strings if you give it an array of bytes that aren't valid UTF-8. That being said, as far as I can tell, that really doesn't matter here as much as it does in a crypto library, so there's really no reason not to just use the latter method.